streams/src/test/java/org/apache/kafka/test/MockProcessorSupplier.java

```diff
@@ -16,30 +16,21 @@
  */
 package org.apache.kafka.test;
 
-import org.apache.kafka.streams.processor.AbstractProcessor;
-import org.apache.kafka.streams.processor.Cancellable;
 import org.apache.kafka.streams.processor.Processor;
-import org.apache.kafka.streams.processor.ProcessorContext;
 import org.apache.kafka.streams.processor.ProcessorSupplier;
 import org.apache.kafka.streams.processor.PunctuationType;
-import org.apache.kafka.streams.processor.Punctuator;
 
 import java.util.ArrayList;
+import java.util.List;
 
 import static org.junit.Assert.assertEquals;
 
 public class MockProcessorSupplier<K, V> implements ProcessorSupplier<K, V> {
```

Comment:
@vvcephei your understanding is right. The ProcessorSupplier it used for the DSL, but we need to create one topology for each task, and hence one processor created for each task. In unit tests thought, most likely only one processor will be created since we will only have one task, unless the same ProcessorSupplier is used in multiple places in the DSL.

Your suggestions on the naming looks good. I will go ahead and make these changes.